napari/utils/perf/_patcher.py

```diff
@@ -0,0 +1,155 @@
+"""Patch callables (functions and methods).
+
+Our perfmon system using this to patch in perf_timers, but this can be used
+for any type of patching. See patch_callables() below as the main entrypoint.
+"""
+
+from importlib import import_module
+import types
+from typing import Callable, List, Set, Tuple, Union
+
+# The parent of a callable is a module or a class, class is of type "type".
+CallableParent = Union[types.ModuleType, type]
+
+# An example PatchFunction is:
+# def _patch_perf_timer(parent, callable: str, label: str) -> None
+PatchFunction = Callable[[CallableParent, str, str], None]
+
+
+class PatchError(Exception):
+    """Failed to patch target, config file error?"""
+
+    def __init__(self, message):
+        self.message = message
+
+
+def _patch_attribute(module, attribute_str, patch_func):
+    """Patch the module's callable pointed to by the attribute string.
```

Comment:
This needs a full docstring.

Reply:
I was sort of a assuming "private" functions/methods did not necessarily need docstrings. Although sometimes I have added them anyway, and I can here. But did I just invent that guideline, does everything need docstrings equally whether public or "private"?

Reply:
I added a Parameters section and type annotations.